dagshub/data_engine/model/datasource.py

```diff
@@ -344,8 +372,11 @@ def func():
             self.source.get_from_dagshub()
             send_analytics_event("Client_DataEngine_addEnrichments", repo=self.source.repoApi)
             ctx = MetadataContextManager(self)
+
+            ctx.start_explicit_ds_ctx()
             yield ctx
-            self._upload_metadata(ctx.get_metadata_entries())
+            self._upload_metadata(ctx.get_metadata_entries() + self._get_source_implicit_metadata_entries())
```

Comment:
You should empty the implicit context on upload finish (maybe in a try/finally block).
Otherwise the values updated in the implicit context will be uploaded again next time you do another `with ds.metadata_context()`.
I guess you can actually repeat the code you did in `save_ctx`, setting the implicit context to None will also work